libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,370 @@
+            maxDate: { type: String, attribute: 'max-date' },
+            minTime: { type: String, attribute: 'min-time' },
+            maxTime: { type: String, attribute: 'max-time' },
+            amPm: { type: String, attribute: 'am-pm' },
+            block: { type: Boolean },
+            readonly: { type: Boolean },
+            error: { type: Boolean },
+            success: { type: Boolean },
+            pattern: { type: String },
+        };
+    }
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [
+            resetStyle,
+            baseStyle,
+            iconStyle,
+            inputFieldStyle,
+            inputAddonStyle,
+            inputGroupStyle,
+            tooltipStyle,
+            datepickerStyle,
+            datepickerUigStyle,
+        ];
+    }
+
+    constructor() {
+        super();
+
+        // Fix tot <-> tot-en-met
+        if (Dutch?.nl) {
+            Dutch.nl.rangeSeparator = ' tot en met ';
+        }
+    }
+
+    protected firstUpdated(changedProperties: Map<string, unknown>): void {
+        super.firstUpdated(changedProperties);
+
+        this.initializeComponent();
+
+        if (!this.isInShadowDOM()) {
+            this.getFlatpickrCalendar()?.classList.add('flatpickr-calendar--push-top');
+        }
+
+        if (changedProperties.has('selectedDate')) {
+            if (this.selectedDate) {
+                const defaultDate = this.parseSelectedDate(this.selectedDate);
+                this.updateValue(defaultDate);
+                this.instance?.setDate(defaultDate, false, this.format);
+            }
+        }
+
+        if (!this.initialValue) {
+            this.initialValue = this.value;
+        }
+    }
+
+    updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+
+        const options = this.getDynamicOptions();
+        this.updateOptionsForInstance(options);
+
+        if (changedProperties.has('value')) {
+            this.updateValue(this.value);
+        }
+
+        if (changedProperties.has('block')) {
+            if (this.block) {
+                this.getFlatpickrWrapper()?.classList.add('flatpickr-wrapper--block');
+            } else {
+                this.getFlatpickrWrapper()?.classList.remove('flatpickr-wrapper--block');
+            }
+        }
+    }
+
+    protected render(): TemplateResult {
+        const inputClasses = {
+            'vl-input-field--error': this.error || this.isInvalid,
+            'vl-input-field--success': this.success,
+            'vl-input-field--block': this.block,
+            'vl-input-field--disabled': this.disabled,
+        };
+
+        const buttonClasses = {
+            'vl-input-addon--error': this.error || this.isInvalid,
+            'vl-input-addon--success': this.success,
+            'vl-input-addon--disabled': this.disabled,
+        };
+
+        return html`
+            <div class="vl-input-group" id="wrapper">
```

Comment:
Is dit id nodig? Indien dit nodig is kan je het een zinvollere naam geven? datepicker-wrapper bv.

Reply:
Zinvollere naam gegeven.